server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -839,6 +839,21 @@ protected String createLocalStoragePoolName(Host host, StoragePoolInfo storagePo
         return String.format("%s-%s-%s", StringUtils.trim(host.getName()), "local", storagePoolInformation.getUuid().split("-")[0]);
     }
 
+    protected void checkNfsOptions(String nfsopts) throws InvalidParameterValueException {
+        String[] options = nfsopts.replaceAll("\\s", "").split(",");
+        Map<String, String> optionsMap = new HashMap<>();
+        for (String option : options) {
+            String[] keyValue = option.split("=");
+            if (keyValue.length > 2) {
+                throw new InvalidParameterValueException("Invalid value for NFS option " + keyValue[0]);
+            }
+            if (optionsMap.containsKey(keyValue[0])) {
+                throw new InvalidParameterValueException("Duplicate NFS option values found for option " + keyValue[0]);
+            }
+            optionsMap.put(keyValue[0], null);
```

Comment:
validation of nfs version and nconnect values, supported nfs versions check needed here?

Reply:
Don't want to perform validations on any mount options as idea is to leave it to the admin if they are using the correct options are not. Cloudstack is only providing the functionality to add options.

We can give an indication that this feature is tested with two mount options (`vers` and `nconnect`). But I am thinking that admin can give even more mount options if they desire to do so. That way we don't have to make any changes for supporting more options.

Reply:
(re-)mount may fail with incorrect options, hope the relevant message is thrown to the operator.

Reply:
That's a good point. Let me test this and get back.

Reply:
I have updated the Testing section in PR description with this testcase.
Had to make some changes in `cancelMaintain()` to note and log the exception passed from libvirt when mount fails.

Reply:
I can add a few lines in the documentation PR on what happens if incorrect mount options are given if you are ok with the current behaviour.